src/Apps/Order/redirects.tsx

```diff
@@ -174,6 +176,43 @@ const goToRespondIfAwaitingBuyerResponse: OrderPredicate = ({ order }) => {
   }
 }
 
+const goToOrder2DetailsIfEnabled: OrderPredicate = ({ order, isEnabled }) => {
+  if (newDetailsEnabled({ order, isEnabled })) {
+    return {
+      path: `/orders2/${order.internalID}/details`,
+      reason: "Order2 is enabled for this order",
+    }
+  }
+}
+
+const goToOrder2CheckoutIfEnabled: OrderPredicate = ({ order, isEnabled }) => {
+  if (newCheckoutEnabled({ order, isEnabled })) {
+    return {
+      path: `/orders2/${order.internalID}/checkout`,
+      reason: "Order2 is enabled for this order",
+    }
+  }
+}
+
+// Temporary type to allow orders queried for the new checkout page to work here
+interface Order2RedirectArgs {
+  order: { mode?: string | null }
+  isEnabled: SystemContextProps["isEnabled"]
+}
+export const newCheckoutEnabled = ({
+  order,
+  isEnabled,
```

Comment:
```
featureFlags
...
return !!(order.mode === "BUY" && featureFlags.isEnabled?.("emerald_checkout-redesign"))
```

Better to keep the temporary function signature clear, since isEnabled` implies a boolean value, vs a function call, where `featureFlags` can point right at the new `FeatureFlags` type